src/commands/azureServiceOperators/helpers/azureservicehelper.ts

```diff
@@ -0,0 +1,287 @@
+export async function installOlm(
+    kubectl: k8s.APIAvailable<any>,
+    clusterKubeConfig: string
+): Promise<k8s.KubectlV1.ShellResult | undefined> {
+    try {
+        const asoOlmYamlFile = "https://github.com/operator-framework/operator-lifecycle-manager/releases/download/v0.17.0/olm.yaml";
+        const runResult = await tmpfile.withOptionalTempFile<k8s.KubectlV1.ShellResult | undefined>(
+            clusterKubeConfig, "YAML",
+            (f) => kubectl.api.invokeCommand(`create -f ${asoOlmYamlFile} --kubeconfig="${f}"`));
+
+        return runResult;
+    } catch (e) {
+        vscode.window.showErrorMessage(`Installing operator lifecycle manager resource had following error: ${e}`);
+        return undefined;
+    }
+}
+
+export async function installOperator(
+    kubectl: k8s.APIAvailable<any>,
+    clusterKubeConfig: string
+): Promise<k8s.KubectlV1.ShellResult | undefined> {
+    try {
+        const asoYamlFile = "https://operatorhub.io/install/azure-service-operator.yaml";
+        const runResult = await tmpfile.withOptionalTempFile<k8s.KubectlV1.ShellResult | undefined>(
+            clusterKubeConfig, "YAML",
+            (f) => kubectl.api.invokeCommand(`create -f ${asoYamlFile} --kubeconfig="${f}"`));
+
+        return runResult;
+    } catch (e) {
+        vscode.window.showErrorMessage(`Installing operator resoruce had following error: ${e}`);
+        return undefined;
+    }
+}
+
+export async function installIssuerCert(
+    kubectl: k8s.APIAvailable<any>,
+    clusterKubeConfig: string
+): Promise<k8s.KubectlV1.ShellResult | undefined> {
+    try {
+        const extensionPath = getExtensionPath();
+        const templateYAML = tmp.fileSync({ prefix: "aso-issuer", postfix: `.yaml` });
+        const yamlPathOnDisk = vscode.Uri.file(path.join(extensionPath!, 'resources', 'yaml', 'issuerandcertmanager.yaml'));
+        const issuerandcertmanager = fs.readFileSync(yamlPathOnDisk.fsPath, 'utf8');
+        fs.writeFileSync(templateYAML.name, issuerandcertmanager);
+
+        const runCommandResult = await tmpfile.withOptionalTempFile<k8s.KubectlV1.ShellResult | undefined>(
+            clusterKubeConfig, "YAML",
+            (f) => kubectl.api.invokeCommand(`apply -f ${templateYAML.name} --kubeconfig="${f}"`));
+
+        return runCommandResult;
+    } catch (e) {
+        vscode.window.showErrorMessage(`ASO Issuer and Cert Deployment file had following error: ${e}`);
+        return undefined;
+    }
+}
+
+export async function installOperatorSettings(
+    kubectl: k8s.APIAvailable<any>,
+    operatorSettingInfo: OperatorSettings,
+    clusterKubeConfig: string
+): Promise<k8s.KubectlV1.ShellResult | undefined> {
+    try {
+        const extensionPath = getExtensionPath();
+        const yamlPathOnDisk = vscode.Uri.file(path.join(extensionPath!, 'resources', 'yaml', 'azureoperatorsettings.yaml'));
+
+        const azureoperatorsettings = fs.readFileSync(yamlPathOnDisk.fsPath, 'utf8');
+        azureoperatorsettings
```

Comment:
Bit confused by this. The `.replace` function returns a _new_ string, it does not modify the string on which it is called. So it looks like all these replacements are bring thrown away. And yet this clearly works when you test it.  What is going on?